src/expr/src/scalar/mod.rs

```diff
@@ -2395,6 +2399,12 @@ impl fmt::Display for EvalError {
             EvalError::InvalidIdentifier { ident, .. } => {
                 write!(f, "string is not a valid identifier: {}", ident.quoted())
             }
+            EvalError::ArrayFillWrongArraySubscripts => {
+                f.write_str("wrong number of array subscripts")
+            }
+            EvalError::MaxArraySizeExceeded(max_size) => {
+                write!(f, "array size exceeds the maximum allowed ({max_size})")
```

Comment:
I think this max size is wrong:
```
materialize=> select array_fill(1, array[26000000])[1];
 array_fill
------------
          1
(1 row)
materialize=> select array_fill(1, array[27000000])[1];
ERROR:  array size exceeds the maximum allowed (134217728)
```

Reply:
Yes, the size is in bytes (not for any particularly interesting reason other than this was PG's limit)--I amended the error message to specify that.